### ISSUE TITLE
Drag'n drop import should create field "created"

### ISSUE THREAD

**Issue by twMat** [Jermolene/TiddlyWiki5#2530]
When I DnD-import a png image, only fields; *title, type, text* are created. I was expecting at least a *created* field... possibly also *modified* which apparently also is default when creating a new tiddler (!?).

This lack also makes the tiddler *not* appear in the Recent list.

**Comment by Jermolene** [comment]
Hi @twMat is this a dupe of #1554?

**Comment by twMat** [comment]
Hm, it did have a familiar feel to it. IMO they are related but separate issues (except my ending note here above). 

Anything that wasn't a tiddler before is IMO certainly getting its "tiddler gestalt" created at the moment of import.

To add a *created* field also does not seem to affect the Recent list; If I understand right, the timeline macro uses *modified* as a default for its *dateField* paramter.... and/but  the Recent list does not provide a *dateField* argument. So an imported tiddler can have/get a *created* field without showing up in the Recent list. (IMO it should show up but that IS issue #1554)

A reason to at least add a *created* field is so one has any time reference to hook onto e.g in filters.

**Comment by tobibeer** [comment]
I'd also think it is a dupe of #1554. Otherwise, what is there to merge from here to there, so you can close one of the two?

**Comment by twMat** [comment]
But if this OP was solved, then issue #1554, i.e if the imported title should appear in Recent, could still be argued. Hence they are different issues IMO.

I rephrased this OP more explicitly.

**Comment by Jermolene** [comment]
@twMat do we really need one ticket to discuss adding a `modified` field to imported tiddlers, and a different one about adding `created` field. I do think it's easier to discuss them as a single issue.

**Comment by twMat** [comment]
@Jermolene - I guess it comes down to if the current behaviour to add neither of *created* nor *modfied* to imported non-tiddlers is a *decision* made at some point rather than an omission or just not yet implemented.

IF it is a decision (presumably of yours) then having  "[all] imports register in Recent" (#1554) might not change the arguments for such a decision, as far as I can see. But then, if solving #1554 would automatically also solve this OP, then great! If you are saying it would, then please close this thread or inform me and I'll close it :-)

Thanks for everyones input!

**Comment by Jermolene** [comment]
> I guess it comes down to if the current behaviour to add neither of created nor modfied to imported non-tiddlers is a decision made at some point rather than an omission or just not yet implemented.

Originally, this behaviour was implemented deliberately: it was to ensure that tiddlers that were imported wouldn't be altered by the importing process.

Then people reported the unexpected behaviour of imported images not showing up in the recent tab. Since then we've been having the various discussions about the issue that you've seen.

There are different aspects to the cases of updating the `modified` and `created` fields for imported tiddlers, but the implementation of both lies in the behaviour of the import mechanism.

**Comment by danielo515** [comment]
> Then people reported the unexpected behaviour of imported images not showing up in the recent tab

Maybe people realized this because of images, but there are some other things that do not include such field and it is expected too. I usually import small text files, for example.

**Comment by twMat** [comment]
The underlying problem in this current issue is poor access to the imported tiddlers - so if [my proposal in #1554](https://github.com/Jermolene/TiddlyWiki5/issues/1554#issuecomment-507074284) or some variant thereof is implemented, i.e to give a retaining view of imported tids, then this issue here will also be solved. (I'm mostly writing this to get a link there.)

**Comment by pmario** [comment]
I think the OP is not the same as issue 1554 especially, if you have a closer look at the mockup. .... It's 1 think to add some new fields to d&d imported content. ... 

IMO it's a completely different thing to create new UI elements during import. 

The first behaviour can be implemented in the core. The second one imo is plugin territory.

**Comment by twMat** [comment]
>I think the OP is not the same as issue 1554 

I agree 100% but, as noted, the underlying problem is poor access to the imported tiddlers.
